### PR TITLE
[JENKINS-67470] Extend shading of Tyrus WebSocket client

### PR DESCRIPTION
**Pull request by jglick**
Solves [JENKINS-67470](https://issues.jenkins-ci.org/browse/JENKINS-67470) by amending #5128. ~Untested. @jlamasrios are you able to verify whether this solves your problem? It was not completely clear to me how to reproduce;~

> ~The problem is also easily reproduced by just trying to create a jakarta.xml.bind.annotation.W3CDomHandler instance from the plugin code.~

~but do you need to have any specific dependency (say on the `jaxb` plugin) for that to happen? Does it happen only on Java 11?~

### Proposed changelog entries

* `ClassNotFoundException: io.jenkins.cli.shaded.org.w3c.dom.Node` when using JAXB.

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Comment by jlamasrios** [comment]
Thanks! I'll try to check on my own scenario and comment here.

> but do you need to have any specific dependency (say on the jaxb plugin) for that to happen? Does it happen only on Java 11?

Our plugin does not depends on jaxb plugin (which uses jaxb 2.3.0 - ie: javax namespace) but directly declares a dependency on the newer 3.0.1 version alternatives (eg: jakarta.xml.bind:jakarta.xml.bind-api, etc.). Not sure if it also fails on Java 8; will try to confirm too.

**Comment by jlamasrios** [comment]
@jglick I can confirm the problem did occur on Java 8 too, and that this change fixes it for both Java 8 and Java 11.

**Review by jlamasrios — APPROVED** [review]
I can confirm the problem did occur on Java 8 too, and that this change fixes it for both Java 8 and Java 11.

**Comment by jglick** [comment]
@basil any opinion given that you are active in #6085 etc.?

**Review by basil — APPROVED** [review]
Shading sounds like the right approach here. This PR is fine as a hotfix that preserves the status quo, but in the future I think this code could be made less fragile if we could find a way to tell Maven to just shade all of Tyrus's dependencies rather than having to provide an explicit list of all of Tyrus's dependencies' packages in our source tree. That list already become out-of-date once (resulting in this bug) and could easily become out-of-date again in the future. Perhaps trading our dependency on `tyrus-standalone-client-jdk` for one on `tyrus-client` might help with that.

**Comment by basil** [comment]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Review by oleg-nenashev — APPROVED** [review]

**Pull request merged**
